libtalek/handle.go

```diff
@@ -142,9 +146,27 @@ func (h *Handle) Decrypt(cyphertext []byte, nonce *[24]byte) ([]byte, error) {
 // sending it to the handle's updates channel if valid.
 func (h *Handle) OnResponse(args *common.ReadArgs, reply *common.ReadReply, dataSize uint) {
 	msg := h.retrieveResponse(args, reply, dataSize)
-	if msg != nil && h.updates != nil {
+	if msg != nil {
 		h.Seqno++
-		h.updates <- msg
+		if h.partialMsgLength == 0 {
```

Comment:
As mentioned earlier, I think we ought to have some notion of a message header. It'd be cleaner to do this as a struct that gets marshalled to a byte array, than to do this with manual byte manipulation.